### PR TITLE
tar: Support completions for zstd compression extensions

### PR DESCRIPTION
**Pull request by gftg85**
(description in the commit message)

### CODE REVIEW

File: completions/tar [hunk 1]
```diff
@@ -433,9 +433,9 @@ __tar_cleanup_prev()
 
 __tar_detect_ext()
 {
-    local tars='@(@(tar|gem|spkg)?(.@(Z|[bgx]z|bz2|lz?(ma|o)))|t@([abglx]z|b?(z)2))'
+    local tars='@(@(tar|gem|spkg)?(.@(Z|[bgx]z|bz2|lz?(ma|o)|zst))|t@([abglx]z|b?(z)2|zst))'
     ext="$tars"
-    regex='\(\(tar\|gem\|spkg\)\(\.\(Z\|[bgx]z\|bz2\|lz\(ma\|o\)\?\)\)\?\|t\([abglx]z\|bz\?2\)\)'
+    regex='\(\(tar\|gem\|spkg\)\(\.\(Z\|[bgx]z\|bz2\|lz\(ma\|o\)\?\|zst\)\)\?\|t\([abglx]z\|bz\?2\)\)'
```
Comment (scop) [inline completions/tar]: Shouldn't we have tzst for regex too?

Comment (gftg85) [reply completions/tar]: You're right. I'll submit a new version. I wonder, though, what test I could have done that would have exposed this omission.

Comment (scop) [reply completions/tar]: See the completions involving listing/completing filenames within archives.

**Comment by scop** [comment]
BTW there was one more affected regex, that's now handled in commit a143560f315ddfbe7c4662cba8a27b93137f425b

**Pull request merged**